plugins/flytekit-duckdb/flytekitplugins/duckdb/task.py

```diff
@@ -21,8 +43,9 @@ class DuckDBQuery(PythonInstanceTask):
     def __init__(
         self,
         name: str,
-        query: Union[str, List[str]],
+        query: Optional[Union[str, List[str]]] = None,
         inputs: Optional[Dict[str, Union[StructuredDataset, list]]] = None,
+        provider: DuckDBProvider = DuckDBProvider.LOCAL,
```

Comment:
Will do!! Thank you.

I will also use `secret_requests` from `PythonAutoContainerTask` and check for it in the constructor rather than adding a new secret argument.